packages/tracing/src/utils.ts

```diff
@@ -73,3 +150,36 @@ export function secToMs(time: number): number {
 
 // so it can be used in manual instrumentation without necessitating a hard dependency on @sentry/utils
 export { stripUrlQueryAndFragment } from '@sentry/utils';
+
+type SentryTracestateData = {
+  trace_id: string;
+  environment?: string;
+  release?: string;
+  public_key: string;
+  user?: { id?: string; segment?: string };
+};
+
+/**
+ * Compute the value of a Sentry tracestate header.
+ *
+ * @throws SentryError (because using the logger creates a circular dependency)
+ * @returns the base64-encoded header value
+ */
+export function computeTracestateValue(data: SentryTracestateData): string {
+  // `JSON.stringify` will drop keys with undefined values, but not ones with null values, so this prevents
+  // these values from being dropped if they haven't been set by `Sentry.init`
+
+  // See https://www.w3.org/TR/trace-context/#tracestate-header-field-values
+  // The spec for tracestate header values calls for a string of the form
+  //
+  //    identifier1=value1,identifier2=value2,...
+  //
+  // which means the value can't include any equals signs, since they already have meaning. Equals signs are commonly
+  // used to pad the end of base64 values though, so to avoid confusion, we strip them off. (Most languages' base64
+  // decoding functions (including those in JS) are able to function without the padding.)
+  try {
+    return unicodeToBase64(JSON.stringify(dropUndefinedKeys(data))).replace(/={1,2}$/, '');
+  } catch (err) {
+    throw new SentryError(`[Tracing] Error computing tracestate value from data: ${err}\nData: ${data}`);
```

Comment:
Hmm this could lead to a spammy stream of errors sent to Sentry if clients are sending bad data. It would be a safer approach here to ignore invalid input, might log the error to the console.

Users can't really fix this error once they see it in Sentry. Either our SDKs have a bug or a bad client is sending data and SDK users can't really fix it on their own.

Reply:
I don't think it'll get through to users' issue streams. Because it's a `SentryError`, it'll just get thrown rather than captured - see https://github.com/getsentry/sentry-javascript/blob/e46ff3e4c819962b045be9325fb1cedaa7178fb0/packages/core/src/baseclient.ts#L546-L548.